sycl/test-e2e/SeparateCompile/sycl-external-within-staticlib.cpp

```diff
@@ -0,0 +1,79 @@
+// Check that, when linking with a static library, SYCL_EXTERNAL device code
+// is preserved despite optimizations.
+// RUN: %{build} -O3 -DSOURCE1 -c -o %t1.o
+// RUN: %{build} -O3 -DSOURCE2 -c -o %t2.o
+// RUN: %{build} -O3 -DSOURCE3 -c -o %t3.o
+// RUN: rm -f %t.a
```

Comment:
Do we need this line because of `%t.a` existing from a previous run of the test? Does `llvm-ar` error if the file already exists?

Reply:
I just copied that part from sycl-external.cpp, but on the first glance, I don't see why it would be needed. I'll check and remove this line from both files in a follow-up PR.

Reply:
It is needed to make sure %t.a is made anew.
Without `rm` and if it exists somehow, objects will be replaced instead of being added.
This may upset something.
There is no error, @sarnex.